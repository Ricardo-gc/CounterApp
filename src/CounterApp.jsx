
export const CounterApp = ( { value } ) => {

    const handleApp = (event, newValue) => {
        console.log(event);
    }
    
    return (
        <>
            <h1>Counter App</h1>
            <h2>{ value }</h2>

            <button onClick={ handleApp }>
                +1    
            </button>
        </>
    );
}