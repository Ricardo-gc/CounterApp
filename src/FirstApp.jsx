import React from 'react'
import PropTypes from 'prop-types';

export const FirstApp = ( {
  title, 
  subTitle,
  name
} ) => {
  return (
    <>
      <h1 data-testid="test-title">{ title }</h1>
      <p>{ subTitle }</p>
      <div>{ name }</div>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string
}

FirstApp.defaultProps = {
  title: 'No title',
  subTitle: 'No SubTitle',
  name: 'Ricardo Garc√≠a'
}

