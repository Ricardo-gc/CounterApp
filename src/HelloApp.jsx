// import React from 'react';

// function App(){
//     // documento.createElement.....
//     return ( <h1>Hello World!!</h1> );
// }

// export default App;
//rafc
import React from 'react'

export const HelloWorldApp = () => {
  return (
    <h1>Hello World App</h1>
  );
}

