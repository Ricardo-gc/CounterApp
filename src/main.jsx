import React from 'react';
import ReactDOM from 'react-dom/client';

import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloApp';
import { CounterApp } from './CounterApp';

// Para importarlo de manera global
import './index.css';

//render el component
ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    {/* <HelloWorldApp /> */}
    {/* <FirstApp title='Hello World!!'/> */}
    <CounterApp value={ 7 }/>
  </React.StrictMode>
)
