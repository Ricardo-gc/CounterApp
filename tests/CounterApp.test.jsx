import { fireEvent, render, screen} from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe(' Pruebas en CounterApp', () => {
    const val = 10;
    test('Should match with snapshot', () => {
        render (<CounterApp value={val}/>);
        expect(screen.container).toMatchSnapshot();
    });
    
    test('Should show initial value of 100', () => {
        render (<CounterApp value={val}/>);
        expect(screen.getByText(val+"")).toBeTruthy();
        expect(screen.getByRole('heading', {level: 2}).innerHTML).toContain(val+"");
    });
    
    test('Should add with button +1', () => {
        render (<CounterApp value={ val }/>);
        //Ejecutar/disparar un evento de click - AÃ±adir 1
        fireEvent.click(screen.getByText('+ 1'))
        expect( screen.getByText( '11' ) ).toBeTruthy();
    });

    test('Should substract with button -1', () => {
        render (<CounterApp value={ val }/>);       
        //Quitar 1
        fireEvent.click(screen.getByText('-1'))
        expect( screen.getByText( '9' ) ).toBeTruthy();
    });
    
    test('Should substract with button reset', () => {
        render (<CounterApp value={ val }/>);       
        //Quitar 1 varias veces
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('-1'))
        fireEvent.click(screen.getByText('-1'))

        //Reset despues de quitar varios
        // fireEvent.click(screen.getByText('Reset'))
        //Para hacerlo mas especifico, revisar el CounterApp para saber el atributo aria-label
        fireEvent.click(screen.getByRole('button', {name: 'btn-reset'}));
        expect( screen.getByText( '10' ) ).toBeTruthy();

    });
});