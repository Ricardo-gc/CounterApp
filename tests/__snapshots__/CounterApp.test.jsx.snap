// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Pruebas en CounterApp Should match with snapshot 1`] = `undefined`;
